Notes application

Lokni opened the pull request.

Lokni committed:
Start new project MyNotes Application.

Lokni committed:
Add tasks and empty fragment for notes list.

Lokni committed:
Add list with example notes in fragment and show them on screen,
made data class StructureData,
add dimensions for text size.

Lokni committed:
Add second activity and fragment set clicklistener on list,
add resources and design for NoteFragment

Lokni committed:
Add second activity and fragment set clicklistener on list,
add resources and design for NoteFragment

Lokni committed:
add HashMap for collecting data, add open new fragment.

Lokni committed:
change linear layout on constraint layout but double line stay

Lokni committed:
change EditText on TextView and fix problem with double line.

Lokni committed:
start work for landscape orientation

Lokni committed:
add landscape orientation without save

Lokni committed:
add save current note, when change orientation.

Lokni committed:
clean and refactor code

Sandor13 commented on AndroidSDK/Basic/Lesson_6/Notes/app/src/main/java/ru/dmkalvan/mynotes/NotesListFragment.java:
```diff
@@ -0,0 +1,34 @@
+        // Required empty public constructor
+    }
+
+    public static NotesListFragment newInstance() {
```
если вы не передаете ничего во Фрагмент, то этот метод не нужен

Lokni committed:
Start home work for lesson 7

Lokni committed:
Recreate app for Single-Activity

Lokni committed:
Add themes noActionBar for menu

Lokni committed:
Add CoordinatorLayout in gradle

Lokni committed:
Add menu toolbar

Lokni committed:
codeCosmetic

Lokni committed:
Start Home work for Lesson 8

Lokni committed:
Add recycler view in fragment_notes_list.xml create template for list element view.

Lokni committed:
Add adapter for Recycler view

Lokni committed:
Add Recycler View in fragment and add clickListener

Lokni committed:
Start problem with:
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: ru.dmkalvan.mynotes, PID: 8409
    java.lang.IllegalArgumentException: No view found for id 0x7f0900a9 (ru.dmkalvan.mynotes:id/fragment_container) for fragment NotesListFragment{60e799e (197fb045-5c6b-4968-9f9c-57308c787ae4) id=0x7f0900a9}

Lokni committed:
Add CardView and change operations with data

DeltAagent00 commented on AndroidSDK/Basic/Lesson_6/Notes/app/src/main/java/ru/dmkalvan/mynotes/NotesListFragment.java:
```diff
@@ -68,7 +77,6 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         dataCollector(2, new StructureData("Wedding Anniversary",
                 "Next month", "15.01.21",
```
Строковые данные нужно выносить в ресурсы приложения

DeltAagent00 commented on AndroidSDK/Basic/Lesson_6/Notes/app/src/main/res/values/strings.xml:
```diff
@@ -4,12 +4,31 @@
+    <string name="date_3">26.01.21</string>
+    <!--  Body  -->
+    <string name="shopping_list_body">1. Potato
+        2. Onion
```
так лучше не делать, либо разделите на разные ресурсы либо расставьте \n (перенос строки)

Lokni committed:
Error with rotation is fixed

Lokni committed:
Add check for orientation in Activity addFragment

Lokni committed:
Start home work for lesson 9

Lokni committed:
Add fragment add note.

Lokni committed:
Add call new fragment

Lokni committed:
add action for save button in AddNoteFragment

Lokni committed:
Add action clear in AddNoteFragment

Lokni committed:
Add context_menu.xml

Lokni committed:
Add context_menu.xml

Lokni committed:
cosmetic

DeltAagent00 commented on AndroidSDK/Basic/Lesson_6/Notes/app/src/main/java/ru/dmkalvan/mynotes/AddNoteFragment.java:
```diff
@@ -47,10 +48,16 @@ public void onCreateOptionsMenu(@NonNull Menu menu, @NonNull MenuInflater inflat
+            case R.id.action_save:
+                if (label.getText() != null || body.getText() != null) {
+                    addDataToList();
+                    Toast.makeText(getContext(), "Note saved", Toast.LENGTH_SHORT).show();
```
Строковые значения нужно выносить в файл переводов

DeltAagent00 commented on AndroidSDK/Basic/Lesson_6/Notes/app/src/main/java/ru/dmkalvan/mynotes/AddNoteFragment.java:
```diff
@@ -31,10 +34,23 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         View view = inflater.inflate(R.layout.fragment_add_note, container, false);
         initView(view);
-        if (body.getText() != null && label.getText() != null){
+        setHasOptionsMenu(true);
```
тут лучше отступить табами, чтобы было видно тело if

Lokni replied:
Да, спасибо, но там нет if

DeltAagent00 commented on AndroidSDK/Basic/Lesson_6/Notes/app/src/main/java/ru/dmkalvan/mynotes/NoteListAdapter.java:
```diff
@@ -52,17 +60,35 @@ public void setOnItemClickListener(OnItemClickListener itemClickListener) {
+
+            card.setOnLongClickListener(v -> {
+                menuPosition = getLayoutPosition();
+                itemView.showContextMenu(10, 10);
```
числа нужно выносить в константы и давать им понятные и однозначные имена

Lokni committed:
fix review noticed
